core/trino-main/src/test/java/io/trino/execution/scheduler/TestFaultTolerantStageScheduler.java

```diff
@@ -933,17 +944,47 @@ private FaultTolerantStageScheduler createFaultTolerantTaskScheduler(
                 ticker,
                 sinkExchange,
                 new FaultTolerantPartitioningScheme(3, Optional.empty(), Optional.empty(), Optional.empty()),
+                sourceExchanges.entrySet().stream().collect(toImmutableMap(Map.Entry::getKey, entry -> {
+                    FaultTolerantStageScheduler sourceScheduler = new FaultTolerantStageScheduler(
```

Comment:
this constructor is somewhat of an abomination :)
Maybe we should have a wrapper `createTestFaultTolerantStageScheduler` which calls the `FaultTolerantStageScheduler()` filling in all the boring args and exposing parameters only for stuff the test code is actually interested in.